### ISSUE TITLE
Plasmashell chashes (when using Qt 5.6)

### ISSUE THREAD

**Issue by vmorenomarin** [Zren/plasma-applets#13]
I have installed the last update of the widget (v25) . When I click over him, Plasmashell chashes and restarts immediately.

I use Chakra OS, with Plasma 5.6.2 over Qt 5.6.0 and KDE Frameworks 5.21.0.

**Comment by Zren** [comment]
Again? fuck. Thought I fixed this in #5 

Did you have it installed before you updated plasma? Which version did you upgrade from if so?

**Comment by vmorenomarin** [comment]
No. I have installed recently the widget.

**Comment by Zren** [comment]
Which version of Chakra OS are you using? Is there a live cd I can test with?

**Comment by vmorenomarin** [comment]
The version of Chakra OS is 4.2.6-1-CHAKRA. And yes, There is a live CD of Chakra. 

I have opened plasmashell through Console to see the outputs. When I click over the widget, this is the output in terminal of this event (the _click_):

```
qml: onMonthViewDateChanged mié. abr. 13 00:00:00 2016 GMT-0500
qml: onSeletedDateChanged mié. abr. 13 00:00:00 2016 GMT-0500
KCrash: Attempting to start /usr/bin/plasmashell from kdeinit
sock_file=/run/user/1000/kdeinit5__0
KCrash: Application 'plasmashell' crashing...
KCrash: Attempting to start /usr/lib/drkonqi from kdeinit
sock_file=/run/user/1000/kdeinit5__0
QSocketNotifier: Invalid socket 8 and type 'Read', disabling...
QSocketNotifier: Invalid socket 28 and type 'Read', disabling...
QSocketNotifier: Invalid socket 10 and type 'Read', disabling...

[1]+  Stopped                 plasmashell
```

**Comment by Zren** [comment]
There won't be any decent stacktraces for this since it's probably calendar backend from the month view from digitalclock/plasma which is written in c++.

I can only see the 2016.02 "ian" ChakraOS livecd which is too old (has kde 5.5). I'll try out the [kubuntu neon cds](http://files.kde.org/neon/neon-devedition-gitstable/current/?) which someone said have 5.6.2 now.

**Comment by vmorenomarin** [comment]
But Chakra is rolling release (well, exactly semi rolling, i. e, updates rapidly Plasma when it has new versions).

But I understand, the iso has that version that you say.

**Comment by Zren** [comment]
Kubuntu Neon seems to work. It's still running 5.5.1 though.

**Comment by vmorenomarin** [comment]
I have installed the widget in my other computer with the same SO, Chakra, with Qt 5.6 and Plasma 5.6.2. It works fine.

**Comment by Zren** [comment]
Hopefully (not) someone else runs into the same crash your main machine is getting, but yeah, until we know how to reproduce it, your out a luck. Thanks for trying the applet out at least.

**Comment by vmorenomarin** [comment]
Finally I could install the widget in my laptop. After several attempts, I decided download the folder of git. I did executed the build script and then the install script that are in the git directory. It works!!!

Excellent work! It's a great widget.

**Comment by fabastrid** [comment]
on Antergos KDE, plasma 5.6.3 keeps crashing

**Comment by vmorenomarin** [comment]
Do you have tried the solution that I have posted? Try downloading the source pkg here in git and install from the downloaded directory.

**Comment by Zren** [comment]
Just want to confirm before I look into this. @fabastrid you're running Qt 5.6? right? `Menu > About System`

Also, do you know if there's a live cd with your particular setup? The last release looks to be from 3.20.

Found these test ISOs https://repo.antergos.info/iso/testing/ so I try them first.

**Comment by rnc** [comment]
@vmorenomarin : I had a crash with the latest Fedora 23 update (which includes QT 5.6) but uninstalling the applet and reinstalling (from github) works flawlessly.

**Comment by fabastrid** [comment]
I have Qt 5.6 yes. Antergos is a pure rolling release, in fact a clone of Arch, so you have always last versions of everything. 
I understand that installing it by git makes it work flawlessly, but why we shouldn't try to solve what is could be just a simple stupid bug?

**Comment by Zren** [comment]
Did you have eventcalendar installed before upgrading to qt 5.6? Or did you install the applet from "addon installer" with qt 5.6 already installed?

**Comment by fabastrid** [comment]
I forgot to tell you, when you try the live, you don't have the packages updated, because the distro updates itself during the installation, so that, at the end of the process all is updated and nothing is to update.
I tried eventcalendar before and after, having crashes anyway.

**Comment by fabastrid** [comment]
I can set eventcalendar up by right click, with my calendar, city, etc., but the left click makes it crash

**Comment by fabastrid** [comment]
Yes, I have Qt 5.6 and that test iso is good. I have no particular setup, just breeze everywhere.

**Comment by Zren** [comment]
Got a link to the livecd iso?

**Comment by fabastrid** [comment]
You mean this one? https://repo.antergos.info/iso/testing/antergos-2016.04.21-x86_64.iso

**Comment by fabastrid** [comment]
But I told you, you can't work on live, you need to install it :(

**Comment by fabastrid** [comment]
A refresh of the iso
https://antergos.com/download/antergos-live-iso-64bit/

**Comment by Zren** [comment]
So the antergos didn't want to run in a VM for me. So I installed ChakraOS and did the updates.

![](https://i.imgur.com/sYMUtTt.png)

I installed the widget through KDE's ui. Plasmashell did restart itself after adding the widget. I'm not 100% sure if that crash was the applets fault or not since I didn't launch plasmashell from the console, so I could read the logs to know for sure. It does appear to run fine afterwards though. Chakra is on KDE 5.6.3 and Q 5.6.0, which is the reported versions.

So does this still crash plasmashell for you guys?
